google/api_core/rest_streaming.py

```diff
@@ -33,7 +31,9 @@ class ResponseIterator:
         class expected to be returned from an API.
 
     Raises:
-        ValueError: If `response_message_cls` is not a subclass of `proto.Message` or `google.protobuf.message.Message`.
+        ValueError:
+            - If `response_message_cls` is not a subclass of `proto.Message` or `google.protobuf.message.Message`.
+            - If `response` is not an instance of `requests.Response`.
```

Comment:
Where is this exception raised? Line 46 will succeed with a matching method of any class, right?

Reply:
I think it's reasonable to not raise an error for this. We've added the relevant type hints. Cleaned up the docstring.